allennlp/data/dataset_readers/dataset_utils/span_utils.py

```diff
@@ -0,0 +1,131 @@
+from typing import List, Tuple, Callable, TypeVar
+
+from allennlp.data.dataset_readers.dataset_utils.ontonotes import TypedStringSpan
+
+T = TypeVar("T")
```

Comment:
What's the reasoning for making this generic?

Reply:
`strings` vs `Tokens` mainly - the only reason I did this is so that it's explicit that the filter_function must be able to deal with the type you provide.

Reply:
Then maybe `T = TypeVar('T', str, Token)`?  And, yeah, that's a good point about the filter function.

Reply:
Oh, I didn't realise you could make things generic but over specific types. Cool, i'll do that.